### PR TITLE
Bug/64/update subjects modules

### PR DESCRIPTION
**Pull request by VoitV**
1. add route categories/names
2. fixed return object for collections
3. add swagger docs for subject

### CODE REVIEW

File: src/modules/subject/subjects.routes.js [hunk 1]
```diff
@@ -11,13 +11,35 @@ const {
 router.use(authMiddleware)
 router.param('id', idValidation)
 
+// @desc    Get all subjects
+// @route 	GET /subject
+// @access  Private (Authenticated users)
 router.get('/', asyncWrapper(subjectController.subjectsFind))
+
+// @desc    Get subject by id
+// @route 	GET /subject/:id
```
Comment (vlad-khrychov) [inline src/modules/subject/subjects.routes.js:20]: it should be 
```suggestion
// @route 	GET /subjects/:id
```

File: src/modules/subject/subjects.routes.js [hunk 2]
```diff
@@ -11,13 +11,35 @@ const {
...
+// @access  Private (Authenticated users)
 router.get('/:id', asyncWrapper(subjectController.subjectFindById))
+
+// @desc    Get all subject by categories id
+// @route 	GET /subject/categories/:id/subjects
```
Comment (vlad-khrychov) [inline src/modules/subject/subjects.routes.js:25]: it should be
```suggestion
// @route 	GET /categories/:id/subjects
```

File: src/modules/subject/subjects.routes.js [hunk 3]
```diff
@@ -11,13 +11,35 @@ const {
...
+// @access  Private (Authenticated users)
 router.get('/categories/:id/subjects', asyncWrapper(subjectController.subjectsFindByCategoryId))
 
 router.use(restrictTo(ADMIN))
+// @desc    Create subject
+// @route 	POST /subject
```
Comment (vlad-khrychov) [inline src/modules/subject/subjects.routes.js]:
```suggestion
// @route 	POST /subjects
```

File: src/modules/subject/subjects.routes.js [hunk 4]
```diff
@@ -11,13 +11,35 @@ const {
 router.use(authMiddleware)
 router.param('id', idValidation)
 
+// @desc    Get all subjects
+// @route 	GET /subject
+// @access  Private (Authenticated users)
 router.get('/', asyncWrapper(subjectController.subjectsFind))
```
Comment (AntonRomanov999) [inline src/modules/subject/subjects.routes.js:17]: is there any router for getting full list of subjects names? as it implemented for categories module

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]
We should change routing in `src/routes/index.js` for subject (rename it into `subject`)   and categories (add route  `/categories/:id/subjects` to this branch).

**Pull request closed without merging**